Locally I am getting (before and after this changes, but with my upstream changes):

{code}
[INFO] --- gwt-maven-plugin:2.8.0:compile (gwt-compile) @ optaplanner-wb-webapp ---
[INFO] Compiling module org.optaplanner.workbench.OptaPlannerWorkbench
[INFO]    Tracing compile failure path for type 'org.optaplanner.core.api.score.AbstractScore'
[INFO]       [ERROR] Errors in 'jar:file:/home/ge0ffrey/projects/jboss/just-optaplanner/optaplanner-wb/optaplanner-wb-upstream-model/target/optaplanner-wb-upstream-model-7.0.0-SNAPSHOT.jar!/org/optaplanner/core/api/score/AbstractScore.java'
[INFO]          [ERROR] Line 186: No source code is available for type org.optaplanner.core.impl.score.director.ScoreDirector<Solution_>; did you forget to inherit a required module?
{code}

My upstream changes didn't change AbstractScore. They did add a method to the Score interface.

--- a/optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/util/PlannerDomainTypes.java
+++ b/optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/util/PlannerDomainTypes.java
@@ -24,6 +24,7 @@ import org.optaplanner.core.api.score.buildin.bendable.BendableScore;
 import org.optaplanner.core.api.score.buildin.bendablebigdecimal.BendableBigDecimalScore;
 import org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScore;
 import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
+import org.optaplanner.core.api.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScore;
 import org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScore;
 import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
 import org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScore;
@@ -47,6 +48,7 @@ public class PlannerDomainTypes {
         SCORE_TYPES.add( BendableBigDecimalScore.class );
         SCORE_TYPES.add( BendableLongScore.class );
         SCORE_TYPES.add( HardMediumSoftScore.class );
+        SCORE_TYPES.add( HardMediumSoftBigDecimalScore.class );
         SCORE_TYPES.add( HardMediumSoftLongScore.class );
         SCORE_TYPES.add( HardSoftScore.class );
         SCORE_TYPES.add( HardSoftBigDecimalScore.class );

--- a/optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/resources/org/optaplanner/workbench/screens/domaineditor/client/resources/i18n/DomainEditorLookupConstants.properties
+++ b/optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/resources/org/optaplanner/workbench/screens/domaineditor/client/resources/i18n/DomainEditorLookupConstants.properties
@@ -18,6 +18,7 @@ BendableScore=Bendable score
 BendableBigDecimalScore=Bendable big decimal score
 BendableLongScore=Bendable long score
 HardMediumSoftScore=Hard medium soft score
+HardMediumSoftBigDecimalScore=Hard medium soft big decimal score
 HardMediumSoftLongScore=Hard medium soft long score
 HardSoftScore=Hard soft score
 HardSoftBigDecimalScore=Hard soft big decimal score

--- a/optaplanner-wb-upstream-model/pom.xml
+++ b/optaplanner-wb-upstream-model/pom.xml
@@ -71,6 +71,7 @@
           <include>org/optaplanner/core/api/score/buildin/bendablebigdecimal/BendableBigDecimalScore.java</include>
           <include>org/optaplanner/core/api/score/buildin/bendablelong/BendableLongScore.java</include>
           <include>org/optaplanner/core/api/score/buildin/hardmediumsoft/HardMediumSoftScore.java</include>
+          <include>org/optaplanner/core/api/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScore.java</include>
           <include>org/optaplanner/core/api/score/buildin/hardmediumsoftlong/HardMediumSoftLongScore.java</include>
           <include>org/optaplanner/core/api/score/buildin/hardsoft/HardSoftScore.java</include>
           <include>org/optaplanner/core/api/score/buildin/hardsoftbigdecimal/HardSoftBigDecimalScore.java</include>

--- a/optaplanner-wb-upstream-model/src/main/resources/ErraiApp.properties
+++ b/optaplanner-wb-upstream-model/src/main/resources/ErraiApp.properties
@@ -34,7 +34,7 @@ errai.marshalling.serializableTypes=org.optaplanner.core.api.score.buildin.benda
     org.optaplanner.core.api.score.buildin.bendablebigdecimal.BendableBigDecimalScore \
     org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScore \
     org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore \
-    org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftBigDecimalScore \
+    org.optaplanner.core.api.score.buildin.hardmediumsoftbigdecimalHardMediumSoftBigDecimalScore \
     org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScore \
     org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore \
     org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScore \

--- a/optaplanner-wb-upstream-model/src/main/resources/ErraiApp.properties
+++ b/optaplanner-wb-upstream-model/src/main/resources/ErraiApp.properties
@@ -34,7 +34,7 @@ errai.marshalling.serializableTypes=org.optaplanner.core.api.score.buildin.benda
     org.optaplanner.core.api.score.buildin.bendablebigdecimal.BendableBigDecimalScore \
     org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScore \
     org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore \
-    org.optaplanner.core.api.score.buildin.hardmediumsoftbigdecimalHardMediumSoftBigDecimalScore \
+    org.optaplanner.core.api.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScore \
     org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScore \
     org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore \
     org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScore \

--- a/optaplanner-wb-upstream-model/src/main/resources/OptaPlannerUpstreamModelSourcePaths.gwt.xml
+++ b/optaplanner-wb-upstream-model/src/main/resources/OptaPlannerUpstreamModelSourcePaths.gwt.xml
@@ -27,7 +27,7 @@
   <source path="org/optaplanner/core/api/score/buildin/bendablebigdecimal" includes="BendableBigDecimalScore.java" />
   <source path="org/optaplanner/core/api/score/buildin/bendablelong" includes="BendableLongScore.java" />
   <source path="org/optaplanner/core/api/score/buildin/hardmediumsoft" includes="HardMediumSoftScore.java" />
-  <source path="org/optaplanner/core/api/score/buildin/hardmediumsoftbigdecima" includes="HardMediumSoftBigDecimalScore.java" />
+  <source path="org/optaplanner/core/api/score/buildin/hardmediumsoftbigdecimal" includes="HardMediumSoftBigDecimalScore.java" />
   <source path="org/optaplanner/core/api/score/buildin/hardmediumsoftlong" includes="HardMediumSoftLongScore.java" />
   <source path="org/optaplanner/core/api/score/buildin/hardsoft" includes="HardSoftScore.java" />
   <source path="org/optaplanner/core/api/score/buildin/hardsoftbigdecimal" includes="HardSoftBigDecimalScore.java" />

